### ISSUE TITLE
Detect usages of default! / null! / Null forgiving operator

### ISSUE THREAD

**Issue by Thieum** [meziantou/Meziantou.Analyzer#1080]
**Is your feature request related to a problem? Please describe.**
Kind of a follow-up of #1047 - Most of the time I encounter a null! / default! in newly written code, it's a code smell, or something that would require a comment to justify the usage of the null forgiving operator.

Others have written better about that than I could:

https://mattjameschampion.com/2025/06/16/the-null-forgiving-operator-in-c-is-a-code-smell/
https://itnext.io/using-the-null-forgiving-operator-in-c-can-be-considered-bad-practice-f68ffc0f8fb9

And your tool suggests that too: https://www.meziantou.net/automating-null-forgiving-operator-removal-in-csharp-projects.htm

**Describe the solution you'd like**
Add an analyzer on the usage of null! / default! or even null forgiving operator if it's simpler and if it makes sense.

**Describe alternatives you've considered**
Nullable.Extended do something similar but is based on the detection of comments that could justify the usage: https://github.com/tom-englert/Nullable.Extended - It seems awkward not to rely on the warning ignore regular mechanism.

**Additional context**

this cases should be detected :

```csharp

HttpClient _httpClient = default!;
string TempDir { get; set; } = null!;
TestId = _model.TestId!.Value,
```

**Comment by meziantou** [comment]
IDE0370 reports all unneeded null-forgiving operator. So, it remains the case where a user knows a property will be assigned to a non-null value before use, but the compiler is not aware of it (e.g. DI, deserialization, value binding).

Do you want to report all `null!` and `default!`? If so, do you need a custom way to document it and remove the warning, or do you plan to use `#pragma warning disable` to document when it's valid?

**Comment by Thieum** [comment]
Reporting all null! and default! would be nice. A `#pragma warning disable` would feel like the standard way to handle and document the exceptions when needed.

**Comment by meziantou** [comment]
As this would be verbose with lots of false-positive in a normal codebase (require manual intervention to disable warning), this rule will be disabled by default (require enable it using editorconfig).
